templates/webpack.config.js

```diff
@@ -1,17 +1,46 @@
-const configFunc = require("@thc/webpack-react/lib/webpack-config");
-const { prepareEntries } = require("@thc/webpack-react/lib/utils");
-//const path = require('path');
+const {
+    handleAssets,
+    handleJs,
+    handleCss,
+    emptyConf,
+    configEntries,
+    addHotReload,
+    addHtmlIndex,
+    miscOptions,
+    optimize,
+    configOutput,
+    generateSourcemap,
+    utilities
+} = require("@thc/webpack-react/lib/blocks");
+
+const { envDefaults } = require("@thc/webpack-react/lib/utils");
 
 module.exports = (processEnv, argv) => {
-    const env = processEnv || process.env;
-    const config = configFunc(env, argv);
+    const env = envDefaults(processEnv);
+    // Every function accepts a config object
+    // Object is gonna be shallow merged with default conf
+    const configFunctions = [
+        emptyConf(), // Must be first
+        handleJs(),
+        handleCss(),
+        handleAssets(),
+        configEntries(), //Default entry object is : { main : "./src/app.js" }
+        addHotReload(),
+        addHtmlIndex(),
+        miscOptions(),
+        optimize(),
+        configOutput(),
+        generateSourcemap(),
+        utilities() // Must be after output => depends on config.output.path
```

Comment:
I'm not found of having `utilities` having a dep on `configOutput`.
It should rely, like `configOutput`, onto `processEnv.OUTPUT_PUBLIC_PATH`

For `hot` or `test`(with value `/\.(js|jsx)$/`) each one had its own config. 
We should keep it this way, or come up with master config.
 But I think it would kind of drift away from what you are trying to accomplish here.